x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/spatial/SpatialPushDownTestCase.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.spatial;
+
+import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.geometry.GeometryCollection;
+import org.elasticsearch.geometry.ShapeType;
+import org.elasticsearch.geometry.utils.WellKnownText;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xpack.core.esql.action.EsqlQueryRequestBuilder;
+import org.elasticsearch.xpack.core.esql.action.EsqlQueryResponse;
+import org.elasticsearch.xpack.esql.plugin.EsqlPlugin;
+import org.elasticsearch.xpack.spatial.SpatialPlugin;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+/**
+ * Base class to check that a query than can be pushed down gives the same result
+ * if it is actually pushed down and when it is executed by the compute engine,
+ *
+ * For doing that we create two indices, one fully indexed and another with index
+ * and doc values disabled. Then we index the same data in both indices and we check
+ * that the same ES|QL queries produce the same results in both.
+ */
+public abstract class SpatialPushDownTestCase extends ESIntegTestCase {
+
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(EsqlPlugin.class, SpatialPlugin.class);
+    }
+
+    /**
+     * Elasticsearch field type
+     */
+    protected abstract String fieldType();
+
+    /**
+     * A random {@link Geometry} to be indexed.
+     */
+    protected abstract Geometry getIndexGeometry();
+
+    /**
+     * A random {@link Geometry} to be used for querying.
+     */
+    protected abstract Geometry getQueryGeometry();
+
+    /**
+     * Necessary to build a ES|QL query. It should be "TO_GEOSHAPE" for geo
```

Comment:
Since the casting operator was added, we can now also do "WKT"::geo_shape

Reply:
It won't work for geo_point

Reply:
`"POINT(0 0)"::geo_point` should work, as would `location::geo_point`. I just tested by editing some tests on 8.15, and it worked.

Reply:
Ah, right, I understand!